chaos-days/blog/2025-07-11-linear-dynamic-scaling/index.md

```diff
@@ -0,0 +1,166 @@
+but only temporarily, as once the scaling operation has completed,
+the additional load it generate is not present anymore.
+
+Then, we will execute the same procedure as above,
+until we hit 2x the critical throughput hit before.
+
+#### Expectation
+
+If the system scales linearly, we expect to see similar level of performance metrics
+for similar values of the ratios `PI (created/complete) per second / nr. of partition`.
+
+## Steady state
+
+The system is started with a throughput of 150 Process instances created per second.
+As this is a **standard benchmark configuration**, nothing unexpected happens:
+
+- The same number of process instances are completed as the ones created
+- The expected number of jobs is completed per unit of time
+
+At this point, we have the following topology:
+
+![initial-topology](./initial-topology.png)
+
+### First benchmark: 3 broker and 3 partitions
+
+Let's start increasing the load incrementally, by adding 30 Process instances/s for every step.
+
+| Time   | Brokers | Partitions | Throughput       | CPU Usage               | Throttling (CPU)        | Backpressure     |
+|--------|---------|------------|------------------|-------------------------|-------------------------|------------------|
+| 09:30  | 3       | 3          | 150 PI/s, 150 jobs/s | 1.28 / 1.44 / 1.02  | 12% / 7% / 1%          | 0                 |
+| 09:49  | 3       | 3          | 180 PI/s, 180 jobs/s | 1.34 / 1.54 / 1.12  | 20% / 17% / 2%         | 0                 |
+| 10:00  | 3       | 3          | 210 PI/s, 210 jobs/s | 1.79 / 1.62 / 1.33  | 28% / 42% / 4%         | 0                 |
+| 10:12  | 3       | 3          | 240 PI/s, 240 jobs/s | 1.77 / 1.95 / 1.62  | 45% / 90% / 26%        | 0/0.5%            |
+
+At 240 Process Instances spawned per second, the system starts to hit the limits:
+![CPU usage @ 240 PI/s](./config_1_240_cpu.png)
+![CPU throttling@ 240 PI/s](./config_1_240_cpu_throttling.png)
+
+And the backpressure is not zero anymore:
+![Backpressure @ 240 PI/s](./config_1_240_backpressure.png)
+
+- The CPU throttling reaches almost 90% on one node (this is probably caused by only one node being selected as **gateway** as previously noted)
+- Backpressure is now constantly above zero, even if it's just 0.5%, it's a sign that we are reaching the throughput limits.
+
+### Second part of the benchmark: scaling to 6 brokers and 6 partitions
+
+With 240 process instances per second being spawned, we send the commands to scale the cluster.
+
+We first scale the `zeebe` *statefulset* to 6 brokers. As soon as the new brokers are running, even before they are healthy,
+we can send the command to include them in the cluster and to increase the number of partition to 6.
+
+This can be done following the [guide in the official docs](https://docs.camunda.io/docs/next/self-managed/zeebe-deployment/operations/cluster-scaling/#2b-scaling-brokers-and-partitions).
```

Comment:
```suggestion
This can be done following the [guide in the official docs](https://docs.camunda.io/docs/next/self-managed/components/orchestration-cluster/zeebe/operations/cluster-scaling/#2b-scaling-brokers-and-partitions).
```